stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -259,61 +347,122 @@ bool swift::addStatusRecord(
   }
 }
 
+static void removeStatusRecordLocked(ActiveTaskStatus status, TaskStatusRecord *record) {
+  bool removedRecord = false;
+  auto cur = status.getInnermostRecord();
```

Comment:
Are we guaranteed that `getInnermostRecord` never returns the record we're searching for? Is it worth a comment to say that it can't be used with that record?

Reply:
Yes - this method is called when we are holding the status record lock so the innermost record should be that lock. Let me add an assert to that end.

Reply:
The existing `assert(removedRecord)` will catch this case, but it won't be very clear why it fired if it does.